### ISSUE TITLE
Allow passing data to sassOptions

### ISSUE THREAD

**Issue by blindfish3** [ls-age/svelte-preprocess-sass#5]
I was trying to pass 'data' to sassOptions and wondering why my component styles were failing to render.  Having dug into the source code it looks like passing 'data' to sassOptions replaces the 'content' being passed from the svelte modules...

Any chance to combine the two; with sassOptions.data prepended to content?

**Comment by LukasHechenberger** [comment]
Hm... What’s your particular use case? Can’t think of any where the same effect cannot be achieved with an `@import '../intro.scss'`.
If we used the *data* instead of the *filename* option to render sass we'd also have to overwrite the *importPath* option to allow relative imports. Sourcemaps would be broken as well.

**Comment by blindfish3** [comment]
I want to be able to set some variables during the build process without having to add an @import to all my components. 'data' was a convenient option for doing that... 
At the least you should document those Sass options that aren't supported and ideally also catch any attempts to pass unsupported options; since at the moment overriding them can break the build.

**Comment by LukasHechenberger** [comment]
I see.
I'll implement it based on your suggestion, but instead of adding `data` to the beginning of the sass input, I'll append it. This way at least line and col numbers are correct in the resulting source map.

**Comment by LukasHechenberger** [comment]
Of course, sass doesn't  support using variables before defining them ;) The `data` option will be appended to the regular styles, just like you suggested, so source map support will be broken. I still recommend to use `@import '../intro.scss' instead, but it doesn't really matter as long as svelte does not use the sourcemaps created by preprocessors...

@blindfish3 Could you review *v0.2.0-beta.0*? (`npm i --save-dev svelte-preprocess-sass@beta`)

**Comment by blindfish3** [comment]
Thanks for the quick update :)
I tried svelte-preprocess-sass@beta and it's working as I'd hoped - i.e.:

- I can set variables directly as strings in `data` and they are applied when my component SASS code is compiled
- I can set an @import in `data` - to import a file where I've set some variables - and again my SASS compiles as I expect

I must admit that, having experimented with it a little, I don't fully understand the intention of the data property (the [documentation](https://www.npmjs.com/package/node-sass#data) is perhaps a little too succinct).  It does work to set variables; but it doesn't appear that styles defined (even in the imported file) get applied - perhaps that's as intended; and not a requirement for me anyway ¯\_(ツ)_/¯

Not thoroughly tested; but I see no obvious impact on sourcemaps: as far as I can see my browser dev tools are pointing me to the correct lines in my source code.

And just to explain the rationale for my request:  I wanted to see whether it was practical to create 'themable' svelte components.  In my components I set some variables with the `!default` flag.  Thanks to your changes I can now apply 'theme files' to override these at build time.  I've yet to determine whether this is a truly practical approach; but I didn't like the idea of components having explicit import requirements just to support theming...

**Comment by jiajing1993** [comment]
Hey @blindfish3 . I am using version 0.20 and this is how I do.
```
style: (function() {
          return sass(
            {
              data: `@import "${__dirname}/src/style/var.scss";`
            },
            { name: "scss" }
          );
        })()
```
But the styling of each component still didnt work. Can you help ?

**Comment by blindfish3** [comment]
@jiajing1993 - looking at the current source code the option I requested didn't make it into the final release of 0.2.0 so it's not currently possible to pass data as described here.
I may have time in the next few days to fork and submit a PR.

**Comment by NicholasPedroso** [comment]
I got this working using rollup and [svelte-preprocess](https://github.com/kaisermann/svelte-preprocess)
```javascript
..
import preprocess from 'svelte-preprocess';

export default {
  ...
  plugins: [
    svelte({
      preprocess: preprocess({
        scss: {
          ...
          data: `@import './src/styles.variables.scss'`;
        }
  ]
}
```
